### PR TITLE
DPDK: Allow branch selection from runbook

### PR DESCRIPTION
**Pull request by mcgov**
Add ability to specify a dpdk branch variable to checkout a git ref for the dpdk repo.

### CODE REVIEW

File: microsoft/testsuites/dpdk/dpdksuite.py [hunk 1]
```diff
@@ -350,7 +358,12 @@ def enable_uio_hv_generic_for_nic(node: Node, nic: NicInfo) -> None:
     )
 
 
-def initialize_node_resources(node: Node, log: Logger, pmd: str) -> DpdkTestResources:
+def initialize_node_resources(
+    node: Node, log: Logger, variables: Dict[str, Any], pmd: str
+) -> DpdkTestResources:
+    assert_that(variables).contains_key("dpdk_branch")
```
Comment (squirrelsc) [inline microsoft/testsuites/dpdk/dpdksuite.py]: It should support a default value. In most test pass, the version should be the default value. The design principle of v3 is to simplify test runs, it means minimize the number of parameters. So change the lines like ,
`dpdk_branch = variables.get("dpdk_branch", "default_branch")`

Comment (mcgov) [reply microsoft/testsuites/dpdk/dpdksuite.py]: I think our default behavior is going to be installing from the package manager. I'll swap this syntax around but the default would still be "" so we can truth check it and default over to the package manager install.

Comment (mcgov) [reply microsoft/testsuites/dpdk/dpdksuite.py]: Also I would like to wait to remove the assert until the package manager install is finished, it's about halfway done rn but will  be the next PR to show up

Comment (mcgov) [reply microsoft/testsuites/dpdk/dpdksuite.py]: Can we merge this? I updated to use the variables.get method, there is a second PR coming that will implement the default path so I'll remove the assert in that PR

Comment (squirrelsc) [reply microsoft/testsuites/dpdk/dpdksuite.py]: The assertion should be removed also, it makes the case failed, if the variable is not provided.

Comment (mcgov) [reply microsoft/testsuites/dpdk/dpdksuite.py]: removing

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**